lightningd/peer_control.c

```diff
@@ -1107,6 +1107,47 @@ peer_connected_hook_deserialize(struct peer_connected_hook_payload *payload,
 	return true;
 }
 
+/* Compare and store `remote_addr` and the `peer_id` that reported it.
+ * If new address was reported by at least one other, do node_announcement */
+static void update_remote_addr(struct lightningd *ld,
+			       struct wireaddr *remote_addr,
+			       struct node_id peer_id)
+{
+	/* init pointers first time */
+	if (ld->remote_addr_v4 == NULL) {
+		ld->remote_addr_v4 = tal(ld, struct wireaddr);
+		*ld->remote_addr_v4 = *remote_addr;
+		ld->remote_addr_v4_peer = peer_id;
+	}
+	if (ld->remote_addr_v6 == NULL) {
+		ld->remote_addr_v6 = tal(ld, struct wireaddr);
+		*ld->remote_addr_v6 = *remote_addr;
+		ld->remote_addr_v6_peer = peer_id;
+	}
```

Comment:
This seems wrong: you set remote_addr_v4 potentially to a v6 address here?

Reply:
Jup, thanks.